cli-output/src/display.rs

```diff
@@ -140,7 +140,7 @@ fn format_account_mode(message: &Message, index: usize) -> String {
         } else {
             "-"
         },
-        if message.is_writable(index) {
+        if message.is_writable(index, /*demote_program_write_locks=*/ true) {
```

Comment:
What if instead we "correct" the message first, then we don't have to modify `is_writable`

Reply:
I guess I'm not excited about mucking with the actual message. It requires changing both the header and potentially re-ordering the account_keys, plus Message/SanitizedMessage is generally not mutable.
Not a big deal just for the CLI, but ickier other places.

Incidentally, a number of these changes, including this one, are identical to how we did it for sysvars: https://github.com/solana-labs/solana/pull/15497

Reply:
I agree, reforming the message isn't ideal, but it saves a lot of work scattered all over and is probably more sound doing it in one place (or a lot fewer places.  It could be part of the `sanitize` step.

Reply:
Hmm, yeah I see your point. Let me see how noodly that approach is.

Reply:
I also prefer to implement this for how #15497 was done because otherwise we need to keep the original tx around to ensure that we record the message as it was signed. Otherwise it won't pass replay.

Reply:
I guess its only temporary, but dang that's a lot of plumbing :-(

Reply:
> otherwise we need to keep the original tx around

Nuts, that's true. I guess we're stuck with plumbing either way, and this is less scary to me.